### ISSUE TITLE
Subcontracting through the factory pattern loses track of context

### ISSUE THREAD

**Issue by mlhaufe** [final-hill/decorator-contracts#236]
Reduced test case:

```ts
const factoryContract = new Contract<Factory>();

@Contracted(factoryContract)
class Factory {
    child(value: string) { return new Child(value); }
    @override
    toString() { throw new TypeError('MUST OVERRIDE'); }
}

const childContract = new Contract<Child>({
    [extend]: factoryContract,
    [invariant](self) { return self.value.length === 1; }
});

@Contracted(childContract)
class Child extends Factory {
    #value: string;

    constructor(value: string) {
        super();
        this.#value = value;
    }

    get value() { return this.#value; }

    @override
    override toString() { return `'${this.#value}'`; }
}

// PASS
test('Construct Child', () => {
    const f = new Factory(),
        c = f.child('a');

    expect(c).toBeInstanceOf(Child);
    expect(c.value).toBe('a');
});

// FAIL
test('Char[toString]()', () => {
    const f = new Factory();
    expect(f.child('a').toString()).toBe('\'a\'');
    expect(() => {
        f.child('abc');
    }).toThrow(AssertionError);
});
```

Calling toString in the second test causes the invariant to fail:

```
TypeError: Cannot read properties of undefined (reading 'length')
```

```ts
[invariant](self) { return self.value.length === 1; }
```

What's strange here is that the getter `value` seems to work (test 1) but `toString` seems to  not be operating on Child.

**Comment by mlhaufe** [comment]
Even further reduced test case:

```ts
// PASS
test('Construct Child', () => {
    const c = new Child('a');

    expect(c).toBeInstanceOf(Child);
    expect(c.value).toBe('a');
});

// FAIL
test('Factory Construct Child', () => {
    const f = new Factory(),
          c = f.child('a');

    expect(c).toBeInstanceOf(Child);
    expect(c.value).toBe('a');
});
```

**Comment by mlhaufe** [comment]
I wonder if a fix for this would also fix #231
